tools/cli/helpers/doc-parser/src/class-doc-parser.php

```diff
@@ -104,15 +146,304 @@ function ( $item ) use ( $ignore ) {
+	 *
+	 * @param string $directory the folder to look in.
+	 *
+	 * @return array an array of filenames.
+	 * @throws \Exception $e If unable to traverse the filesystem.
+	 */
+	public function get_wp_files( $directory ) {
+		$iterableFiles = new \RecursiveIteratorIterator(
+			new \RecursiveDirectoryIterator( $directory )
+		);
+		$files         = array();
+
+		try {
+			foreach ( $iterableFiles as $file ) {
+				if ( 'php' !== $file->getExtension() ) {
+					continue;
+				}
+
+				$files[] = $file->getPathname();
+			}
+		} catch ( \UnexpectedValueException $exc ) {
+			throw new \Exception( sprintf( 'Directory [%s] contained a directory we can not recurse into', $directory ) );
+		}
+
+		return $files;
+	}
+
+	/**
+	 * Returns an array of docblock annotations for apply_filter function calls, with keys being names of filters
+	 * used.
+	 *
+	 * @param array $nodes  Parser node objects for hook calls.
+	 * @return array docblock annotations.
+	 */
+	public function get_filter_calls( $nodes ): array {
+
+		$blocks = array();
+
+		foreach ( $nodes as $node ) {
+
+			$arguments = $node->getArgs();
+			$hook_name = array_shift( $arguments );
+
+			// Purging any comments that could have been attributed to this argument.
+			$comments = $hook_name->getComments();
+			if ( ! empty( $comments ) ) {
+				$hook_name->setAttribute( 'comments', null );
+			}
+
+			$new_block = array(
+				'type'      => $node->name->name === 'apply_filters' ? 'filter' : 'action',
+				'line'      => $node->getLine(),
+				'end_line'  => $node->getEndLine() > 0 ? $node->getEndLine() : $node->getLine(),
+				'name'      => $this->pretty_print_hook_name( $hook_name ),
+				'arguments' => array(),
+			);
+
+			foreach ( $arguments as $argument ) {
+				$new_block['arguments'][] = $this->printer->prettyPrint( array( $argument ) );
+			}
+			$blocks[] = $new_block;
+		}
+
+		return $blocks;
+	}
+
+	/**
+	 * Pretty prints the name for the hook, taking an argument object as input.
+	 *
+	 * @suppress PhanRedundantConditionInLoop Phan is having trouble getting some of nikic/php-parser type definitions.
+	 *
+	 * @param Node\Arg $argument the first argument to the apply_filter or do_action call.
+	 * @return String pretty printed argument name.
+	 * @throws \Exception On an unexpected argument component.
+	 */
+	public function pretty_print_hook_name( Node\Arg $argument ): string {
+
+		if (
+			$argument->value instanceof Node\Scalar\String_
+				|| $argument->value instanceof Node\Expr\ConstFetch
+				|| $argument->value instanceof Node\Expr\ClassConstFetch
+		) {
+			return trim( $this->printer->prettyPrint( array( $argument ) ), '\'' );
+
+		} elseif ( $argument->value instanceof Node\Scalar\InterpolatedString ) {
+
+			$value = $argument->value;
+			'@phan-var Node\Scalar\InterpolatedString $value';
+
+			$result = '';
+
+			$parts = $value->parts;
+			'@phan-var array<string,Node\InterpolatedStringPart> $parts';
```

Comment:
This one also seems unnecessary. And wrong, since apparently it's more like
```suggestion
			'@phan-var (Expr|Node\InterpolatedStringPart)[] $parts';
```